File: sistemas/mobile/ohara/src/app/model/avaliacao.ts
import { Usuario } from "./usuario";

export class Avaliacao {
    idUsuario: number | null;
    usuario: Usuario | null;
    idManga: number | null;
    nota: number;

    constructor() {
        this.idUsuario = 0;
        this.usuario = new Usuario();
        this.idManga = 0;
        this.nota = 0;
    }
}

File: api/ohara/src/main/java/br/com/ohara/model/Avaliacao.java
package br.com.ohara.model;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class Avaliacao {
	@NotNull
	private Long idUsuario;
	
	private Usuario usuario;
	
	@NotNull
	private Long idManga;
	
	@Min(value = 0, message = "A nota não deve ser menor que zero")
	@Max(value = 5, message = "A nota não deve ser maior que cinco")
	private double nota;
}
